lib/utils/visUtil.hpp

```diff
@@ -527,6 +529,112 @@ class movingAverage {
     bool initialised = false;
 };
 
+// The following class is modified based on
+// https://www.nayuki.io/page/sliding-window-minimum-maximum-algorithm
+/**
+ * @class SlidingWindowMinMax
+ *
+ * @brief Use two deques to keep tracking minimum and maximum values.
+ **/
```

Comment:
Keep all the comments in the same block.  Doxygen allows for extended description details in the main body of the comment block. 
```suggestion
/**
 * @class SlidingWindowMinMax
 *
 * @brief Use two deques to keep tracking minimum and maximum values.
 *
 * This class is based on a modified version of:
 * https://www.nayuki.io/page/sliding-window-minimum-maximum-algorithm
 **/
```